### ISSUE TITLE
UWP USB with Ronin-S

### ISSUE THREAD

**Issue by Noemata** [Wodie/RoninS_CAN#1]
Perhaps we can help each other.  I'd like to build a UWP app that controls the device.  I'm open to suggestions on how to get protocol details.

**Comment by ceinem** [comment]
Hi @Noemata 
I'm trying to get the CAN interface of the RS2 working, in case you are interested in collaborating

**Comment by Wodie** [comment]
> Hi @Noemata
> I'm trying to get the CAN interface of the RS2 working, in case you are interested in collaborating

RS2 CAN interface is public. What kind of collaboration you are looking for?

**Comment by Wodie** [comment]
> Perhaps we can help each other. I'd like to build a UWP app that controls the device. I'm open to suggestions on how to get protocol details.

Debugging or reverse engineering is your only way AFAIK.

**Comment by ceinem** [comment]
> > Hi @Noemata
> > I'm trying to get the CAN interface of the RS2 working, in case you are interested in collaborating
> 
> RS2 CAN interface is public. What kind of collaboration you are looking for?

The documentation is public, but so far, I couldn't find anyone who had successfully integrated it in a project. I'm working on creating a ROS interface for it to use the RS2 in robotics research.

**Comment by Noemata** [comment]
I was hoping there would be overlap between how RS2 CAN works and the WiFi side of things, but given the very different nature of the two, clearly there is none.  I'd like to be able to duplicate what the DJI apps do over WiFi.  I've looked into WiFi sniffers; not in my budget.  It's so unfortunate to have gear like this looked in.  Any ideas?

**Comment by ceinem** [comment]
yeah, a mobile SDK for the Ronin Gimbals was a discussion point on the dji forums for years. unfortunate that nothing has happened in that realm since then. Could you run Android and the DJI App in a VM and try to sniff the network traffic going in and out of the vm to the gimbal? Just as a thought. 
Anyway, why does the connection to the gimbals have to be so complicated? Gimbals by Gremsy for example have open UART ports and perfectly documented SDKs. But they are also quite a bit more expensive than the DJI ones.

**Comment by Noemata** [comment]
@ceinem , agreed.  Trouble is, the Ronin series are Gimbals are top notch.  That's an interesting idea I hadn't considered, using a VM.
